### PR TITLE
Adding CVE-2024-28859 for SwiftMailer gadget chain in Symfony 1.x

### PR DESCRIPTION
**Pull request by darkpills**
There is a deserialization gadget chain in SwiftMailer that impacts Symfony 1.x.
Adding this in both SwiftMailer and Symfony directories

### CODE REVIEW

File: symfony/symfony/CVE-2024-28859.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+title:     Deserialization Gadget chain in Swift Mailer dependancy
+link:      https://github.com/FriendsOfSymfony1/symfony1/security/advisories/GHSA-wjv8-pxr6-5f4r
+cve:       CVE-2024-28859
+branches:
+    1.x:
+        time:     2024-02-27 20:26:56
+        versions: ['>=1.3.0', '<1.15.18']
+reference: composer://symfony/symfony
```
Comment (xabbuh) [inline symfony/symfony/CVE-2024-28859.yaml:8]:
```suggestion
reference: composer://friendsofsymfony1/symfony1
```

(and the filename needs to changed accordingly)

File: swiftmailer/swiftmailer/CVE-2024-28859.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    4.x:
+        time:     null
+        versions: ['>=4.0.0']
+    5.x:
+        time:     null
+        versions: ['>=5.0.0']
+    6.x:
+        time:     2020-12-8 19:18:59
+        versions: ['>=6.0.0', '<6.30.0']
+reference: composer://swiftmailer/swiftmailer
```
Comment (xabbuh) [inline swiftmailer/swiftmailer/CVE-2024-28859.yaml:14]:
```suggestion
reference: composer://friendsofsymfony1/swiftmailer
```

(and the filename needs to be changed accordingly)

**Review by xabbuh — CHANGES_REQUESTED** [review]

File: swiftmailer/swiftmailer/CVE-2024-28859.yaml (hunk 2 above)
Comment (darkpills) [reply swiftmailer/swiftmailer/CVE-2024-28859.yaml:14]: Hello, @xabbuh , thank for you checks. `reference` for the file `swiftmailer/swiftmailer/CVE-2024-28859.yaml` corresponds to `swiftmailer/swiftmailer` repository. But it is archived and there has been a fork. So I suggest maybe making 2 files for referencing both repositories?

**Comment by darkpills** [comment]
`php validator.php` fails asking for upper bounds to the version number but there is no upper bound since the issue is not resolved and won't be on `swiftmailer/swiftmailer` repository at least. How do you solve this situation?

**Comment by thirsch** [comment]
> `php validator.php` fails asking for upper bounds to the version number but there is no upper bound since the issue is not resolved and won't be on `swiftmailer/swiftmailer` repository at least. How do you solve this situation?

Hi @darkpills, maybe just adding the latest release as the upper bound? As it is unlikely that there will come a new release in the archived repository. But I'm not sure, if it's the way to go here.

File: symfony/symfony/CVE-2024-28859.yaml (hunk 1 above)
Comment (thirsch) [reply symfony/symfony/CVE-2024-28859.yaml]: @darkpills as symfony1 was never installable via composer, maybe it's correct to only have one file for symfony1 but keep the approach of two files for swiftmailer?

**Comment by darkpills** [comment]
@thirsch  I added upper bounds and aligned file path with composer path. I also added another CVE. I think it should be ok now.

**Comment by thePanz** [comment]
Thank you @darkpills for taking care of this :+1:

**Comment by stof** [comment]
an unpatched vulnerability is supported. But you should have `null` as the `time` as there would be not timestamp for the release of the patch.

File: friendsofsymfony1/swiftmailer/CVE-2024-28859.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+title:     Deserialization Gadget chain in Swift Mailer
+link:      https://github.com/FriendsOfSymfony1/symfony1/security/advisories/GHSA-wjv8-pxr6-5f4r
+cve:       CVE-2024-28859
+branches:
+    4.x:
+        time:     null
+        versions: ['>=4.0.0', '<=4.3.1']
+    5.x:
+        time:     null
```
Comment (stof) [inline friendsofsymfony1/swiftmailer/CVE-2024-28859.yaml:9]: this time should be the time of the backport done in https://github.com/FriendsOfSymfony1/swiftmailer/pull/18 (the time of the merge to be exact)

File: swiftmailer/swiftmailer/CVE-2024-28859.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+title:     Deserialization Gadget chain in Swift Mailer
+link:      https://github.com/FriendsOfSymfony1/symfony1/security/advisories/GHSA-wjv8-pxr6-5f4r
+cve:       CVE-2024-28859
+branches:
+    4.x:
+        time:     null
+        versions: ['>=4.0.0', '<=4.3.1']
+    5.x:
+        time:     null
+        versions: ['>=5.0.0', '<=5.4.12']
```
[the remark this answers is not in the document]
Comment (darkpills) [reply swiftmailer/swiftmailer/CVE-2024-28859.yaml]: @stof  thank you, i made the required changes

**Review by stof — APPROVED** [review]

**Pull request merged**